calico_cni_k8s_test.go

```diff
@@ -1055,8 +1055,8 @@ var _ = Describe("CalicoCni", func() {
 					// Now create a K8s pod.
 					name := fmt.Sprintf("run%d", rand.Uint32())
 
-					cniContainerIDX := "container-id-00X"
```

Comment:
@robbrockbank is there a reason we need to limit container IDs to be lowercase?

Probably fine, but might bite us some day.

Reply:
container ID is used in the WEP name construction.  We could certainly make the fields more permissive and just have the name construction lowercase it.